graph-builder/src/registry.rs

```diff
@@ -191,107 +208,116 @@ pub async fn fetch_releases<S: ::std::hash::BuildHasher>(
-        // Try to read the architecture from the manifest
-        let arch = match manifest.architectures() {
-            Ok(archs) => {
-                // We don't support ManifestLists now, so we expect only 1
-                // architecture for the given manifest
-                ensure!(
-                    archs.len() == 1,
-                    "[{}] broke assumption of exactly one architecture per tag: {:?}",
-                    tag,
-                    archs
-                );
-                archs.first().map(std::string::ToString::to_string)
-            }
-            Err(e) => {
-                error!(
-                    "could not get architecture from manifest for tag {}: {}",
-                    tag, e
-                );
-                None
-            }
-        };
+            // Replace the tag specifier with the manifestref
+            release.source = {
+                let mut source_split: Vec<&str> = release.source.split(':').collect();
+                let _ = source_split.pop();
 
-        let layers_digests = manifest
-            .layers_digests(arch.as_ref().map(String::as_str))
-            .map_err(|e| format_err!("{}", e))
-            .context(format!(
-                "[{}] could not get layers_digests from manifest",
-                tag
-            ))?
-            // Reverse the order to start with the top-most layer
-            .into_iter()
-            .rev()
-            .collect();
-
-        let mut release = match cache_release(
-            layers_digests,
-            authenticated_client.to_owned(),
-            registry.host.to_owned().to_string(),
-            repo.to_owned(),
-            tag.to_owned(),
-            cache,
-        )
-        .await?
-        {
-            Some(release) => release,
-            None => {
-                // Reminder: this means the layer_digests point to layers
-                // without any release and we've cached this before
-                continue;
-            }
-        };
+                format!("{}@{}", source_split.join(":"), manifestref)
+            };
 
-        // Replace the tag specifier with the manifestref
-        release.source = {
-            let mut source_split: Vec<&str> = release.source.split(':').collect();
-            let _ = source_split.pop();
+            // Attach the manifestref this release was found in for further processing
+            release
+                .metadata
+                .metadata
+                .insert(manifestref_key.to_owned(), manifestref);
+
+            // Process the manifest architecture if given
+            if let Some(arch) = arch {
+                // Encode the architecture as SemVer information
+                release.metadata.version.build =
+                    vec![semver::Identifier::AlphaNumeric(arch.to_string())];
+
+                // Attach the architecture for later processing
+                release.metadata.metadata.insert(
```

Comment:
TODO: double `metadata` is confusing, this should be reworked (in a follow up PR)?

Reply:
Not in scope for this PR, as the code you're commenting on is not actually new but just changed its position and indentation.